tests/test_config_loader.py

```diff
@@ -446,6 +447,38 @@ def test_assign_null(
         assert cfg == {"normal_yaml_config": True, "abc": None}
         assert len(recwarn) == 0
 
+    def test_sweep_config_cache(self, path: str) -> None:
+        setup_globals()
+
+        config_loader = ConfigLoaderImpl(
+            config_search_path=create_config_search_path(path)
+        )
+        master_cfg = config_loader.load_configuration(
+            config_name="config.yaml",
+            strict=False,
+            overrides=["time='${now:%H-%M-%S}'", "home='${env:HOME}'"],
+            run_mode=RunMode.RUN,
+        )
+
+        # trigger resolution by type assertion
+        assert type(master_cfg.time) == str
+        assert type(master_cfg.home) == str
+
+        master_cfg_cache = OmegaConf.get_cache(master_cfg)
+        assert "now" in master_cfg_cache.keys() and "env" in master_cfg_cache.keys()
+        assert master_cfg.home == os.environ["HOME"]
+
+        sweep_cfg = config_loader.load_sweep_config(
+            master_config=master_cfg,
+            sweep_overrides=["time='${now:%H-%M-%S}'", "home='${env:HOME}'"],
+        )
+
+        sweep_cfg_cache = OmegaConf.get_cache(sweep_cfg)
+        assert len(sweep_cfg_cache.keys()) == 1 and "now" in sweep_cfg_cache.keys()
+        assert sweep_cfg_cache["now"] == master_cfg_cache["now"]
+        os.environ["HOME"] = "/another/home/dir/"
```

Comment:
This is now changing the HOME environment variable for the rest of the tests.

Use [monkeypatch](https://docs.pytest.org/en/stable/monkeypatch.html#monkeypatching-environment-variables) to undo it automatically.

Reply:
Good catch and thanks for the pointer! Addressed in latest commit